source/base/function_signed_distance.cc

```diff
@@ -245,7 +278,59 @@ namespace Functions
       while (std::abs(delta_t) > tolerance && iter < max_iter);
       AssertIndexRange(iter, max_iter);
 
-      const double distance = std::hypot(x - px, y - py);
+      AssertIsFinite(x);
+      AssertIsFinite(y);
+
+      return Point<dim>(sign_px * x + center[0], sign_py * y + center[1]);
```

Comment:
Maybe easier to read:
```suggestion
      return (center + Point<dim>(sign_px * x, sign_py * y));
```